client/src/js/components/bhCurrencySelect.js

```diff
@@ -97,8 +97,6 @@ function bhCurrencySelect($scope, Currencies) {
 
     // if the two array lengths are equal, it means every currency is disabled
     $ctrl.allDisabled = ($ctrl.currencies.length === array.length);
-    if ($ctrl.allDisabled) {
-      $ctrl.form.$setValidity('currencies', false);
-    }
+    $ctrl.form.currency.$setValidity('currency', !$ctrl.allDisabled);
```

Comment:
Is there a reason for always setting the validity like this? 

There is no reason for this to block the pull request however it is much harder to understand what this one liner is doing vs. the more explicit - if everything is disable - set validity to false.